### PR TITLE
RSP-1700: Update TooltipTrigger with context

### PR DESCRIPTION
**Pull request by pst67662**
Remove click as a trigger option to tooltips
	

- Remove press handling from `useTooltipTrigger` and `PressResponder` from `TooltipTrigger`

IgnoreEmulatedMouseEvents in `useHover`
	

- Add onTouchStart handler in pointer events fallback code, and set ignoreEmulatedMouseEvents flag to true

	

- In onMouseEnter and onMouseLeave check ignoreEmulatedMouseEvents and ignore the event if it is true.

	

- Reset ignoreEmulatedMouseEvents to false after onMouseLeave

	

- NOTE: iOS 13 implements pointer events, so this code path won't be hit. Test by commenting out the pointer event handling code.

Update useTooltipTrigger

- 	Remove tooltipProps since they are unused
- 	Pass triggerRef instead of triggerProps
- 	Pass state as second arg instead of object property

Add focus handling to `useTooltipTrigger`

Make all state changes go through `TooltipManager`
	

- e.g. onKeyDown currently updates it's state directly, but doesn't update the manager.

Create a FocusableContext
	

- something like this: https://github.com/adobe-private/react-spectrum-v3/compare/focusable-context

`TooltipTrigger` should pass `triggerProps` through the `FocusableContext` instead of `DOMPropsResponder`{{}}

Tooltips will be focus and hover only.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/focus/src/useFocusable.tsx [hunk 1]
```diff
@@ -12,17 +12,53 @@
...
+  children?: ReactNode,
+  ref?: MutableRefObject<HTMLElement>
+}
+
+export const FocusableContext = React.createContext<FocusableContextProps>(null);
+
+export function useFocsuableContext(ref: RefObject<HTMLElement>): FocusableContextProps {
+  let context = useContext(FocusableContext) || {} as FocusableContextProps;
+
+  useEffect(() => {
```
Comment (pst67662) [inline packages/@react-aria/focus/src/useFocusable.tsx:33]: have the same problem with ref as DOMPropsContext https://github.com/adobe-private/react-spectrum-v3/blob/master/packages/%40react-aria/interactions/src/DOMPropsContext.ts#L36

File: packages/@react-aria/focus/src/useFocusable.tsx [hunk 2]
```diff
@@ -12,17 +12,53 @@
...
 }
 
+interface FocusableContextProps extends HTMLAttributes<HTMLElement> {
+  children?: ReactNode,
+  ref?: MutableRefObject<HTMLElement>
+}
+
+export const FocusableContext = React.createContext<FocusableContextProps>(null);
+
+export function useFocsuableContext(ref: RefObject<HTMLElement>): FocusableContextProps {
```
Comment (LFDanLu) [inline packages/@react-aria/focus/src/useFocusable.tsx:30]: 'useFocsuableContext' -> 'useFocusableContext'

File: packages/@react-aria/focus/src/useFocusable.tsx [hunk 3]
```diff
@@ -12,17 +12,53 @@
...
+    <FocusableContext.Provider value={context}>
+      {children}
+    </FocusableContext.Provider>
+  );
+});
+
 export function useFocusable(props: FocusableProps, domRef?: RefObject<HTMLElement>) {
   let {focusProps} = useFocus(props);
   let {keyboardProps} = useKeyboard(props);
+  let domProps = useFocsuableContext(domRef);
```
Comment (LFDanLu) [inline packages/@react-aria/focus/src/useFocusable.tsx:61]: update to 'useFocusableContext'

File: packages/@react-aria/tooltip/src/useTooltip.ts [hunk 4]
```diff
@@ -25,14 +25,13 @@ export function useTooltip(props: SpectrumTooltipProps): TooltipAria {
     role = 'tooltip'
   } = props;
 
-  let {onPointerLeave, onPointerEnter} = useContext(DOMPropsResponderContext) || {};
+  let focusableProps = useContext(FocusableContext);
```
Comment (LFDanLu) [inline packages/@react-aria/tooltip/src/useTooltip.ts:28]: minor thing, but perhaps you could use `useFocusableContext()` instead of `useContext` here?

**Comment by pst67662** [comment]
> Just some small suggestions, otherwise looks good to me.
> 
> Note: the tooltip gets closed when you click the tooltip trigger the first time but this is something that exists with the current implementation so it doesn't have to block this PR

fixed in https://github.com/adobe-private/react-spectrum-v3/pull/469

File: packages/@react-aria/tooltip/src/useTooltip.ts (hunk 4 above)
Comment (pst67662) [reply packages/@react-aria/tooltip/src/useTooltip.ts:28]: useFocusableContext requires ref as argument (useTooltip does not have ref)... other way is to make ref as optional argument in useFocusableContext, if thats ok?

**Comment by snowystinger** [comment]
Was incorporated into https://github.com/adobe/react-spectrum/pull/843 Thanks @pst67662 !!!!!

**Pull request closed without merging**